### ISSUE TITLE
amplify mock function does not install python dependencies

### ISSUE THREAD

**Issue by emdv88** [aws-amplify/amplify-cli#9975]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### Amplify CLI Version

7.6.19

### What operating system are you using?

macOS 12.2.1

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

no

### Amplify Categories

function

### Amplify Commands

Not applicable

### Describe the bug

this is a duplicate of closed bug report #7924 since it was not fixed and I am still running into the same issue.

I try to mock a python lambda function which uses the requests library.

I added import requests to the generated src/index.py.

I modified the generated Pipfile to include

```
[packages]
src = {editable = true, path = "./src"}
requests = "==2.27.1"
```

Then I ran:
amplify mock function myfunction

### Expected behavior

I expect my function to be run, including the `import requests` statement.

### Reproduction steps

1. amplify init
1. amplify add function
1. pipenv shell
1. pipenv install requests==2.27.1
1. modify index.py to import requests
1. amplify mock function myfunction

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>
Below is the complete output of the `amplify mock function` command:

```
Ensuring latest function changes are built...
Installing dependencies from Pipfile.lock (577196)...
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 0/0 — 00:00:00
Starting execution...
Warning: the environment variable LANG is not set!
We recommend setting this in ~/.profile (or equivalent) for proper expected behavior.
Creating a virtualenv for this project...
Pipfile: /Users/werk/Projects/BlueWasp/Pelican1/components/frontend/amplify/backend/function/myfunction/Pipfile
Using /Users/werk/.virtualenvs/myfunction-VVsTcE9B/bin/python3.8 (3.8.11) to create virtualenv...
⠙ Creating virtual environment...created virtual environment CPython3.8.11.final.0-64 in 864ms
  creator CPython3Posix(dest=/Users/werk/.local/share/virtualenvs/myfunction-VVsTcE9B, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/Users/werk/Library/Application Support/virtualenv)
    added seed packages: pip==22.0.4, setuptools==60.9.3, wheel==0.37.1
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator

✔ Successfully created virtual environment! 
Virtualenv location: /Users/werk/.local/share/virtualenvs/myfunction-VVsTcE9B
Traceback (most recent call last):
  File "/Users/werk/.amplify/lib/amplify-python-function-runtime-provider/shim/shim.py", line 35, in <module>
    main(sys.argv[1:])
  File "/Users/werk/.amplify/lib/amplify-python-function-runtime-provider/shim/shim.py", line 22, in main
    spec.loader.exec_module(handlerModule)
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/werk/Projects/BlueWasp/Pelican1/components/frontend/amplify/backend/function/myfunction/src/index.py", line 1, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
myfunction failed with the following error:
Error: Command failed with exit code 1: pipenv run python3 /Users/werk/.amplify/lib/amplify-python-function-runtime-provider/shim/shim.py /Users/werk/Projects/BlueWasp/Pelican1/components/frontend/amplify/backend/function/myfunction/src/index.py handler
Warning: the environment variable LANG is not set!
We recommend setting this in ~/.profile (or equivalent) for proper expected behavior.
Creating a virtualenv for this project...
Pipfile: /Users/werk/Projects/BlueWasp/Pelican1/components/frontend/amplify/backend/function/myfunction/Pipfile
Using /Users/werk/.virtualenvs/myfunction-VVsTcE9B/bin/python3.8 (3.8.11) to create virtualenv...
⠙ Creating virtual environment...created virtual environment CPython3.8.11.final.0-64 in 864ms
  creator CPython3Posix(dest=/Users/werk/.local/share/virtualenvs/myfunction-VVsTcE9B, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/Users/werk/Library/Application Support/virtualenv)
    added seed packages: pip==22.0.4, setuptools==60.9.3, wheel==0.37.1
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator

✔ Successfully created virtual environment! 
Virtualenv location: /Users/werk/.local/share/virtualenvs/myfunction-VVsTcE9B
Traceback (most recent call last):
  File "/Users/werk/.amplify/lib/amplify-python-function-runtime-provider/shim/shim.py", line 35, in <module>
    main(sys.argv[1:])
  File "/Users/werk/.amplify/lib/amplify-python-function-runtime-provider/shim/shim.py", line 22, in main
    spec.loader.exec_module(handlerModule)
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/werk/Projects/BlueWasp/Pelican1/components/frontend/amplify/backend/function/myfunction/src/index.py", line 1, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
    at makeError (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/execa/index.js:118:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at pythonInvoke (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-python-function-runtime-provider/src/util/invokeUtil.ts:33:22)
    at timeConstrainedInvoker (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-util-mock/src/func/index.ts:67:12)
    at start (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-util-mock/src/func/index.ts:48:20)
    at Object.run (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-util-mock/src/commands/mock/function.ts:14:5)
    at Object.executeAmplifyCommand (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-util-mock/src/amplify-plugin-index.ts:8:3)
    at executePluginModuleCommand (/usr/local/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:178:3)
    at executeCommand (/usr/local/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:30:5)
    at Object.run (/usr/local/lib/node_modules/@aws-amplify/cli/src/index.ts:205:5) {
  shortMessage: 'Command failed with exit code 1: pipenv run python3 /Users/werk/.amplify/lib/amplify-python-function-runtime-provider/shim/shim.py /Users/werk/Projects/BlueWasp/Pelican1/components/frontend/amplify/backend/function/myfunction/src/index.py handler',
  command: 'pipenv run python3 /Users/werk/.amplify/lib/amplify-python-function-runtime-provider/shim/shim.py /Users/werk/Projects/BlueWasp/Pelican1/components/frontend/amplify/backend/function/myfunction/src/index.py handler',
  escapedCommand: 'pipenv run python3 "/Users/werk/.amplify/lib/amplify-python-function-runtime-provider/shim/shim.py" "/Users/werk/Projects/BlueWasp/Pelican1/components/frontend/amplify/backend/function/myfunction/src/index.py" handler',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: 'Warning: the environment variable LANG is not set!\n' +
    'We recommend setting this in ~/.profile (or equivalent) for proper expected behavior.\n' +
    'Creating a virtualenv for this project...\n' +
    'Pipfile: /Users/werk/Projects/BlueWasp/Pelican1/components/frontend/amplify/backend/function/myfunction/Pipfile\n' +
    'Using /Users/werk/.virtualenvs/myfunction-VVsTcE9B/bin/python3.8 (3.8.11) to create virtualenv...\n' +
    '\r⠋ Creating virtual environment...\b\r⠙ Creating virtual environment...\b\r⠹ Creating virtual environment...\b\r⠸ Creating virtual environment...\b\r⠼ Creating virtual environment...\b\r⠴ Creating virtual environment...\b\r⠦ Creating virtual environment...\b\r⠧ Creating virtual environment...\b\r⠇ Creating virtual environment...\b\r⠏ Creating virtual environment...\b\r⠋ Creating virtual environment...\b\r⠙ Creating virtual environment...created virtual environment CPython3.8.11.final.0-64 in 864ms\n' +
    '  creator CPython3Posix(dest=/Users/werk/.local/share/virtualenvs/myfunction-VVsTcE9B, clear=False, no_vcs_ignore=False, global=False)\n' +
    '  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/Users/werk/Library/Application Support/virtualenv)\n' +
    '    added seed packages: pip==22.0.4, setuptools==60.9.3, wheel==0.37.1\n' +
    '  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator\n' +
    '\n' +
    '\b✔ Successfully created virtual environment! \n' +
    'Virtualenv location: /Users/werk/.local/share/virtualenvs/myfunction-VVsTcE9B\n' +
    'Traceback (most recent call last):\n' +
    '  File "/Users/werk/.amplify/lib/amplify-python-function-runtime-provider/shim/shim.py", line 35, in <module>\n' +
    '    main(sys.argv[1:])\n' +
    '  File "/Users/werk/.amplify/lib/amplify-python-function-runtime-provider/shim/shim.py", line 22, in main\n' +
    '    spec.loader.exec_module(handlerModule)\n' +
    '  File "<frozen importlib._bootstrap_external>", line 843, in exec_module\n' +
    '  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed\n' +
    '  File "/Users/werk/Projects/BlueWasp/Pelican1/components/frontend/amplify/backend/function/myfunction/src/index.py", line 1, in <module>\n' +
    '    import requests\n' +
    "ModuleNotFoundError: No module named 'requests'",
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
Finished execution.
```

</details>

### Additional information

From what I can tell, the requests package is available in the "source" virtualenv at 
`/Users/werk/.virtualenvs/myfunction-VVsTcE9B/bin/python3.8/site-packages`
but it's not being installed into the target virtualenv at 
`/Users/werk/.local/share/virtualenvs/myfunction-VVsTcE9B`,

**Comment by emdv88** [comment]
Note that when I `amplify push` the whole thing and test it on the AWS console, it does work just fine.

**Comment by ykethan** [comment]
hey @emdv88, could you please try CD into the lambda function folder and run "pipenv --venv".

additionally, we can check the requests library has been loaded in the virtual environment by utilizing the following:

1. Activate the virtual env with "pipenv shell"
2. Enter the REPL with python
3. Then import requests

as this will allow us to check if the library was installed in the right location.

I tried replicating the steps in my local environment as well but was unable to replicate the error. Please do revert back to us we would be happy in digging deeper.

**Comment by danielleadams** [comment]
Hi @emdv88, have you asked this issue in the Amplify [Discord server](https://discord.gg/invite/amplify)? It's possible that another developer that uses Amplify has run into this issue and may be able to help.

**Comment by ykethan** [comment]
hey @emdv88, just wanted to check in to see if you are facing any continued issues with your python lambda functions.

**Comment by josefaidt** [comment]
Closing due to inactivity. @emdv88 if you are still experiencing this issue please reply back to this thread and we can re-open to investigate further 🙂

**Comment by emdv88** [comment]
@josefaidt  and others, excuse the late answer. Turns out I don't get emails on replies in this issue. 

I assume that the issue is still around but for other reasons I pulled my lambda function out of the amplify app and thereby I don't experience it anymore.

My suspicion is that this has to do with me using virtualenvwrapper and hence having the project virtualenv not inside a venv folder in my project folder. Instead it resides in `~/.virtualenvs/<virtualenv name>`. I did however not track this down further.
